src/app/dev/DevToys.Blazor/Components/UIElements/UIMultiLineTextInputPresenter.razor.cs

```diff
@@ -178,6 +180,54 @@ private Task OnMonacoEditorInitializedAsync()
         return Task.CompletedTask;
     }
 
+    private async Task OnMonacoEditorPositionChanged(CursorPositionChangedEvent cursorPosition)
+    {
+        if (UIMultiLineTextInput.Tooltips.Count == 0)
+        {
+            return;
+        }
+
+        await _monacoInitializationAwaiter.Task;
+        TextModel model = await _monacoEditor.GetModelAsync();
+        if (model is not null && cursorPosition.Position is not null)
+        {
+            WordAtPosition wordAtPosition = await model.GetWordAtPositionAsync(JSRuntime, cursorPosition.Position);
+            if (wordAtPosition is not null)
+            {
+                UIHoverTooltip? tooltip = UIMultiLineTextInput.Tooltips.FirstOrDefault(tooltip => tooltip.Word.Equals(wordAtPosition.Word, StringComparison.OrdinalIgnoreCase));
+                if (tooltip != null)
+                {
+                    var decoration = new ModelDeltaDecoration
+                    {
+                        Range = new Range
+                        {
+                            StartLineNumber = 1,
+                            StartColumn = 1,
+
+                            EndLineNumber = wordAtPosition.EndColumn,
+                            EndColumn = wordAtPosition.EndColumn
+                        },
+                        Options = new ModelDecorationOptions
+                        {
+                            HoverMessage = new[]
+                            {
+                                new MarkdownString()
+                                {
+                                    Value = tooltip.Description
+                                }
+                            }
+                        }
+                    };
+                    await _monacoEditor.ReplaceAllDecorationsByAsync(new[] { decoration });
```

Comment:
I think I understand why you made it this way specifically for JWT. You get the list of claims a JWT, you get the localized description for it and try to display it at a tooltip. But you don't know where the claim is in the JSON. So you rely on the current word position.
It has a few drawback though:
1. You can to click on a word in order to see the tooltip.
2. With the current implementation, I may see the tooltip on another word (if I don't click but move the mice), which is confusing.
3. This Tooltip thing is part of the public API. I personally suspect developers will want to display tooltip based on a position rather than a word. What about a sentence for example? Or a whole line? Or a whole paragraph?

**Here is a suggestion:**
Instead of setting one single tooltip when the caret change, would it be possible to:
1. Change `UIHoverTooltip.Word` to `UIHoverTooltip.Span`, of type `TextSpan`. This way, we can set a tooltip on a span instead of a word.
2. Listen to `IUIMultiLineTextInput.HoverTooltipChanged`, so we apply the tooltip when the developer set them.
3. When the HoverTooltip change, for each tooltip:
    1. Translate the position of `UIHoverTooltip.Span` into a Monaco position (line and column) using [GetPositionAtAsync](https://github.com/veler/DevToys/blob/1f1670a73fd6f2983d6d3dff22cae6d0ab641744/src/app/dev/DevToys.Blazor/Components/Text/MonacoEditor/Monaco/TextModel.cs#L192-L193) (that's assuming we change `UIHoverTooltip.Word` into a `TextSpan`). This way, you can end up with a Monaco's `Range`.
    2. Create an instance of `ModelDeltaDecoration` for each tooltip, and pass them to `ReplaceAllDecorationsByAsync`.

Finally, in the JWT tool, you can look at the list of Claims you got, look on each line whether `"iut"` (with quotes) exist. If yes, then you know that you can display the tooltip at that place (or on the whole line, assuming the JSON is formatted).
It requires a little bit more computation, but the user experience and API would be nice.

What do you think?